tests/integration/security/reachability/mtls_test.go

```diff
@@ -0,0 +1,192 @@
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            aApp,
+					To:              headlessApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-permissive.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-off.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "single-port-mtls-on.yaml",
+			namespace:  appsInstance.Namespace().Name(),
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+	}
+
+	for _, c := range testCases {
+		namespace := c.namespace
+		if len(namespace) == 0 {
+			namespace = istioCfg.SystemNamespace
+		}
+
+		policy := policy.ApplyPolicyFile(t, env, namespace, c.configFile)
+		defer policy.TearDown()
+		// Give some time for the policy propagate.
+		// TODO: query pilot or app to know instead of sleep.
+		time.Sleep(time.Second)
+		for _, conn := range c.connections {
+			retry.UntilSuccessOrFail(t, func() error {
+				return connect.CheckConnection(t, conn)
+			}, retry.Delay(time.Second), retry.Timeout(10*time.Second))
+		}
+	}
+}
+
+// TestAuthentictionPermissiveE2E these cases are covered end to end
```

Comment:
why is this commented out? Does it not work?

Reply:
Done: these are comments from some older files. I have revised the comments to explain the tests in this file.